### ISSUE TITLE
gender recognition based on the whole body

### ISSUE THREAD

**Issue by kclch** [matterport/Mask_RCNN#909]
Can I use gender recognition based on the whole body?

**Comment by fastlater** [comment]
It can. Why not?
If you have the masks, you can train. It is a difficult case and you will need a large dataset. Will you try to recognise genders in naked people or dressed people? I guess recognising genders in naked people will be easier since body shapes will be clearly and noticeably different.

**Comment by kclch** [comment]
Mask rcnn takes people as a class. I want to make a large dataset of men and women, and use the source code of mask rcnn to divide people into male and female. Is this method feasible? Will it not be distinguished?

**Comment by fastlater** [comment]
In theory, MaskRCNN is feasible for this kind of task. However, I assume that you will need a very large dataset of bodies.

**Comment by kclch** [comment]
![1](https://user-images.githubusercontent.com/39482063/45333895-e8d2d880-b5aa-11e8-80a5-f6fd4401bebc.png)
Similar to this kind of data set, using labelme to label the dataset as both male and female labels, is this method feasible?

**Comment by kclch** [comment]
![1](https://user-images.githubusercontent.com/39482063/45333999-68f93e00-b5ab-11e8-9782-3d668b13f008.png)
Similar to this kind of dataset, there are about 40,000 images, including one person's different perspective images. Using labelme to label the dataset as male and female labels, is this feasible?

**Comment by fastlater** [comment]
> If human can do it, computer can. Andrew Ng.

That was the reason I asked if you gonna recognize if bodies are naked or wearing clothes.
For me, it is very difficult to say it is a woman for sure I didn't see the pony tail.
You can try but I feel that results wont be too good.
I guess face detection will get better results. Just saying.

**Comment by kclch** [comment]
Thank you very much for your answer. Some people have done gender recognition based on face. I want to use gender recognition based on pedestrians (wearing clothes). Do you have any recommended method? Thank you very much.

**Comment by fastlater** [comment]
Not really. 
As I said, I dont think it will be easy to notice that difference between men and women but that is just my personal opinion.
I guess there have to be some works on the web where you can get info and maybe guide you into a good way to train the network.

**Comment by kclch** [comment]
@fastlater If so, do I need to label the male and female datasets with labelme?

**Comment by fastlater** [comment]
No idea how gonna be better to approach the problem but definitely it will be very challenging to train properly the network. Better read more papers so you can find some hints. I still believe body shape and clothes are not good features that can help you to notice if someone is a woman or a man so maybe you will need a huge dataset with high quality images. But that is just my opinion.

**Comment by Mps24-7uk** [comment]
@kclch Share the dataset  I want to give a shot.

**Comment by ajay960singh** [comment]
@kclch I'm working on a similar project - to detect gender from full body images. Could you get feasible results from your model? Also, can you share your dataset?Thanks

**Comment by Aavesh** [comment]
@ajay960singh  @kclch  hey guys have u been able to make any progress. I am also looking to build something similar

**Comment by ajay960singh** [comment]
@Aavesh No, I'm still trying to collect a good dataset. How far are you in the process?

**Comment by Aavesh** [comment]
@ajay960singh  i am in search , will update here if i make any progress.

**Comment by jamessmith90** [comment]
@fastlater 
As he said i have also tried multiple models but none of them seem to give even decent accuracy. The output is just very random.

**Comment by htso** [comment]
@kclch @ajay960singh @jamessmith90 My thought is while a single snapshot, or pose of a person is hard to tell, a sequence of these would give enough hint. Love to hear any comment and update.

**Comment by balavenkatesh3322** [comment]
@kclch Actually, I want gender recognition based on the whole body. so how does gender classification works using face? Is there any algorithm is available?

**Comment by balavenkatesh3322** [comment]
> > If human can do it, computer can. Andrew Ng.
> 
> That was the reason I asked if you gonna recognize if bodies are naked or wearing clothes.
> For me, it is very difficult to say it is a woman for sure I didn't see the pony tail.
> You can try but I feel that results wont be too good.
> I guess face detection will get better results. Just saying.

@fastlater How gender classification works using face image? could you pls explain little about the core concept?

**Comment by Ritik-Khudiya** [comment]
@kclch can you send me dataset of human whole body

**Comment by ramafahad** [comment]
anyone got a dataset? I want to train yolov11
